File: c/03-estrutura-repetitiva/senha_fixa.c
/*Problema "senha_fixa" (adaptado de URI 1114)
Escreva um programa que repita a leitura de uma senha até que ela seja válida. Para cada leitura de
senha incorreta informada, escrever a mensagem "Senha Invalida! Tente novamente:". Quando a senha
for informada corretamente deve ser impressa a mensagem "Acesso Permitido" e o algoritmo
encerrado. Considere que a senha correta é o valor 2002.*/

#include <stdio.h>
int main () {
    const int correct_password = 2002;
    int password; 

    printf("Digite a senha: ");
    scanf("%d", &password);

    while (password != correct_password) {
        printf("Senha Invalida! Tente novamente: ");
        scanf("%d", &password);
    }

    printf("Acesso Permitido\n");
    return 0;
}

File: java/03-estruturas-repetitivas/Senha_fixa.java
/*Problema "senha_fixa" (adaptado de URI 1114)
Escreva um programa que repita a leitura de uma senha até que ela seja válida. Para cada leitura de
senha incorreta informada, escrever a mensagem "Senha Invalida! Tente novamente:". Quando a senha
for informada corretamente deve ser impressa a mensagem "Acesso Permitido" e o algoritmo
encerrado. Considere que a senha correta é o valor 2002.*/

import java.util.Scanner;
import java.util.Locale;

public class Senha_fixa {

	public static void main(String[] args) {
		Locale.setDefault(Locale.US);
		Scanner sc = new Scanner(System.in);
		
		final int correct_password= 2002;
		int password;
		
		System.out.print("Digite a senha: ");
		password = sc.nextInt();
		
		while (password != correct_password) {
			System.out.print("Senha Invalida! Tente novamente: ");
			password = sc.nextInt();
		}
		
		System.out.println("Acesso permitido!");
		
		sc.close();
	}

}
